Mathlib/Topology/SubsetProperties.lean

```diff
@@ -2041,3 +2041,32 @@ theorem IsPreirreducible.preimage {Z : Set α} (hZ : IsPreirreducible Z) {f : β
 #align is_preirreducible.preimage IsPreirreducible.preimage
 
 end Preirreducible
+
+section codiscrete
+
+/-- The empty subset of any topological space is a discrete set in its
+subspace topology. -/
+lemma discreteTopology_emptyset : DiscreteTopology ↑(∅ : Set α) :=
+  discreteTopology_iff_nhds.mpr (fun x ↦ False.elim x.property)
```

Comment:
I'm surprised this is needed. Doesn't [Subsingleton.discreteTopology](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Order.html#Subsingleton.discreteTopology) apply?